tests/helics/shared_library/badInputTests.cpp

```diff
@@ -100,3 +100,904 @@ TEST(error_tests, unavailable_core_type)
+
+    auto subid = helicsFederateRegisterTypeInput(vFed1, "inp1", "custom2", "", nullptr);
+
+    helicsInputAddTarget(subid, "pub1", nullptr);
+
+    auto inp3 = helicsFederateGetSubscription(vFed1, "unknown", &err);
+    EXPECT_NE(err.error_code, 0);
+    helicsErrorClear(&err);
+    EXPECT_EQ(inp3, nullptr);
+
+    helicsFederateSetTimeProperty(vFed1, helics_property_time_period, 1.0, nullptr);
+
+    helicsFederateEnterInitializingModeAsync(vFed1, &err);
+    EXPECT_EQ(err.error_code, 0);
+    helicsFederateEnterInitializingModeComplete(vFed1, &err);
+    EXPECT_NE(err.error_code, 0);
+    helicsErrorClear(&err);
+
+    helicsFederateFinalize(vFed1, nullptr);
+}
+
+TEST_F(function_tests, initError4)
+{
+    SetupTest(helicsCreateValueFederate, "test", 1);
+    auto vFed1 = GetFederateAt(0);
+    // register the publications
+
+    helicsFederateRegisterGlobalTypePublication(vFed1, "pub1", "custom1", "", nullptr);
+
+    auto subid = helicsFederateRegisterTypeInput(vFed1, "inp1", "custom2", "", nullptr);
+
+    helicsInputAddTarget(subid, "pub1", nullptr);
+
+    helicsFederateSetTimeProperty(vFed1, helics_property_time_period, 1.0, nullptr);
+
+    helicsFederateEnterExecutingModeAsync(vFed1, &err);
+    EXPECT_EQ(err.error_code, 0);
+
+    helicsFederateEnterExecutingModeComplete(vFed1, &err);
+    EXPECT_NE(err.error_code, 0);
+    helicsErrorClear(&err);
+
+    helicsFederateIsAsyncOperationCompleted(vFed1, &err);
+    EXPECT_EQ(err.error_code, 0);
+
+    helics_iteration_result res;
+    helicsFederateRequestTimeIterative(
+        vFed1, 1.0, helics_iteration_request_no_iteration, &res, &err);
+    EXPECT_NE(err.error_code, 0);
+    helicsErrorClear(&err);
+    EXPECT_EQ(res, helics_iteration_result_error);
+
+    helicsFederateFinalize(vFed1, nullptr);
+
+    helicsFederateEnterExecutingModeAsync(vFed1, &err);
+    EXPECT_NE(err.error_code, 0);
+    helicsErrorClear(&err);
+
+    helicsFederateEnterExecutingModeIterativeAsync(
+        vFed1, helics_iteration_request_force_iteration, &err);
+    EXPECT_NE(err.error_code, 0);
+}
+
+TEST_F(function_tests, initError5)
+{
+    SetupTest(helicsCreateValueFederate, "test", 1);
+    auto vFed1 = GetFederateAt(0);
+    // register the publications
+
+    helicsFederateRegisterGlobalTypePublication(vFed1, "pub1", "custom1", "", nullptr);
+
+    auto subid = helicsFederateRegisterTypeInput(vFed1, "inp1", "custom2", "", nullptr);
+
+    helicsInputAddTarget(subid, "pub1", nullptr);
+
+    helicsFederateSetTimeProperty(vFed1, helics_property_time_period, 1.0, nullptr);
+
+    helicsFederateEnterExecutingModeIterative(vFed1, helics_iteration_request_no_iteration, &err);
+    EXPECT_NE(err.error_code, 0);
+    helicsErrorClear(&err);
+
+    helicsFederateRequestTimeIterativeAsync(
+        vFed1, 1.0, helics_iteration_request_no_iteration, &err);
+    EXPECT_NE(err.error_code, 0);
+    helicsErrorClear(&err);
+
+    helicsFederateFinalize(vFed1, nullptr);
+}
+
+TEST_F(function_tests, initError6)
```

Comment:
This is true of most of the tests, but having a comment at the start of the test summarizing what it is testing would make identifying why a future change makes the test start failing easier (and if the test is testing what it claims).